### DISCUSSION TITLE
Exposing files in the result folder

### DISCUSSION THREAD

**Discussion by gregdenay** [bio-raum/FooDMe2#14] [General]
@marchoeppner 

How do you declare which files will be moved to `results` ? I am trying to add the bitscore-processed BLAST results and remove the raw BLAST but I can't seem to make sense this.

**Comment by marchoeppner** [comment]
Right - in principle, you want to look here:

https://github.com/bio-raum/FooDMe2/blob/dev/conf/modules.config

This goes by the name of the processes as included in your respective workflow (i.e. we can rename processes on include, as discussed).

This will then store all files that are **declared as outputs** by the process in the folder specified in the publishDir block:

```NEXTFLOW
withName: BLAST_FILTER_BITSCORE {
        ext.args = "--bit_diff ${params.blast_bitscore_diff}"
        publishDir = [
            path: { "${params.outdir}/${meta.sample_id}/blast/raw" },
            mode: params.publish_dir_mode,
            enabled: true,
            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
        ]
    }
```

The operative option there is "enabled" - set it to true to emit the results, and to false if you want to suppress the results. 

As an added "gottcha", there is a universal declaration at the top which stores all results that do NOT have a process-specific rule in a folder named after the module

```NEXTFLOW
publishDir = [
        path: { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
        mode: params.publish_dir_mode,
        enabled: true,
        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
    ]
```
So just deleting a process "rule" wouldn't work and instead dump the results in a folder named after that process. Hence why "enabled: false" would be needed to do it in a clean way.

**Reply by gregdenay** [reply]
As a good snakemaker I didn't think to look in `conf` thx
